packages/core/src/config/parse.ts

```diff
@@ -506,29 +533,48 @@ export const parseInplaceBytes: VariableHandler<UserConfigVariable, string> = (
 ): string => {
   const { variable, variableType, storageObj } = props
 
+  return parseBytes(
+    variable,
+    variableType.label,
+    storageObj.label,
+    variableType.numberOfBytes
+  )
+}
+
+/**
+ * Handles parsing and validating fixed size bytes for both variables and constructor args.
+ *
+ * @param variable Variable to parse.
+ * @param label Label to use in error messages.
+ * @returns DataHexString
+ */
+const parseBytes = (
+  variable: UserConfigVariable,
+  variableType: string,
+  label: string,
+  numberOfBytes: number
+) => {
   // Check that the user entered a string
   if (typeof variable !== 'string') {
     throw new InputError(
-      `invalid input type for ${
-        storageObj.label
-      }: ${variable}, expected DataHexString but got ${stringifyVariableType(
+      `invalid input type for ${label}: ${variable}, expected DataHexString but got ${stringifyVariableType(
         variable
       )}`
     )
   }
 
-  if (variableType.label.startsWith('bytes')) {
+  if (variableType.startsWith('bytes')) {
     // hexDataLength returns null if the input is not a valid hex string.
     if (ethers.utils.hexDataLength(variable) === null) {
```

Comment:
`isHexString(variable)` doesn't work for this because it will return true if the variable is a valid hex string (which could be an odd number of characters i.e `0x123`). We need to check if the variable is a `DataHexString` with an even length (i.e 0x1234).

Reply:
I'm a little confused. I meant we should replace `ethers.utils.hexDataLength(variable) === null` with `ethers.utils.isHexString(variable)`

Reply:
`ethers.utils.isHexString(variable)` isn't sufficient for this. It checks if the input string is a hex string, but we need to check if it is a DataHexString. DataHexStrings are not equivalent to hex strings. For example, `0x123` is a hex string, but it is not a `DataHexString` because 3 is not a complete two-character byte. On the other hand, `0x1234` is a valid DataHexString because 34 is a valid two-character byte. If we use `ethers.utils.isHexString(variable)` here, then we will not detect invalid inputs such as `0x123` which can lead to errors. 

You can try this out yourself by using `ethers.utils.isHexString(variable)` here and then updating `bytesTest` in the `tests/constants.ts` to `0x123` and running `yarn test:hardhat`. You'll see that while the contract is successfully deployed, but the value 0x123 is not set as expected.

Reply:
`bytesTest` is a dynamic bytes type, and this section of the code is for fixed-length bytes. It appears that this `hexDataLength` check in `parseBytes` is unnecessary because of the [if-statement immediately after it](https://github.com/chugsplash/chugsplash/blob/a0a454323ab769d39828bd64dbbe91eed958430e/packages/core/src/config/parse.ts#L575-L580). If you remove the `hexDataLength` if-statement and enter incorrect inputs for a fixed-length bytes variables, like `bytes32Test`, you'll see that the `isHexString(variable, length)` clause catches everything. So, it seems like we should either remove the `hexDataLength(...) === null` check, or we should change it to `isHexString(variable)` if we want to throw a slightly more helpful error message when the user enters an incorrectly formatted bytes variable (e.g. they entered a `Uint8Array` or forgot the `0x` prefix).